### ISSUE TITLE
[Bug]: v1.33.1-319-nightly very high disk usage

### ISSUE THREAD

**Issue by zeylos** [netdata/netdata#12715]
### Bug description

Hi there,

I have multiple nodes using `v1.33.1-319-nightly` on a child-parent architecture (Kubernetes cluster, one parent and 3/4 childs).
With this version appeared a bug where the child is using a lot of disk IO and triggers the `Disk backlog` alert among others →
![image](https://user-images.githubusercontent.com/24226163/163962795-ad7561fe-32db-4f97-a4d0-1d7cfbf69137.png)
![image](https://user-images.githubusercontent.com/24226163/163962980-145d3ccf-8ade-4b56-a2a5-2e156c8f36f1.png)
Looking at `iotop` on the node, we can clearly see that netdata is the issuer
![image](https://user-images.githubusercontent.com/24226163/163965247-8ccb45da-b2fc-41f0-bf65-576b6ac28389.png)
Looking at the logs of the child, I can see these logs that I have never seen before
![image](https://user-images.githubusercontent.com/24226163/163963810-e514287d-513f-4fa7-9a8a-67255ba1c95b.png)

This is a bug that happened on 4 childs on 3 different Kubernetes clusters in 4 days. This is causing HighDiskPressure on the Kubernetes node, which then stops scheduling pods. 

Killing the child so it restarts is the only way I found to solve the issue for now.

### Expected behavior

netdata not using all my IOs :D

### Steps to reproduce

No idea, this even happened on nodes that I didn't look up on cloud for a week, so it seems that it's not linked to Cloud usage.

### Installation method

helmchart (kubernetes)

### System info

```shell
# uname -a; grep -HvE "^#|URL" /etc/*release
Linux <redacted>.compute.internal 5.4.156-83.273.amzn2.x86_64 #1 SMP Sat Oct 30 12:59:07 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
/etc/os-release:NAME="Amazon Linux"
/etc/os-release:VERSION="2"
/etc/os-release:ID="amzn"
/etc/os-release:ID_LIKE="centos rhel fedora"
/etc/os-release:VERSION_ID="2"
/etc/os-release:PRETTY_NAME="Amazon Linux 2"
/etc/os-release:ANSI_COLOR="0;33"
/etc/os-release:CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
/etc/system-release:Amazon Linux release 2 (Karoo)
```

### Netdata build info

```shell
bash-5.1# netdata -W buildinfo
Version: netdata v1.33.1-319-nightly
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--without-bundled-protobuf' '--with-bundled-libJudy' '--disable-ebpf' 'CFLAGS=' 'LDFLAGS='
Install type: oci
    Binary architecture: x86_64
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    NO
    IPMI:                    YES
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 YES
    Prometheus Remote Write: YES
```

**Comment by MrZammler** [comment]
Hey @zeylos ! So after a restart, it's ok for a while until at some point it starts pushing the disk? Is it possible to correlate that point in time with logs from the agent that's causing it somehow? I.e. if you can send the part of the logs the time it starts?

Also, is it possible to check the size of the sqlite database ? Thank you!

**Comment by zeylos** [comment]
hey @MrZammler, it stops for good, I'll keep a eye and tell you if it does restart pushing the disk. 

About the logs, the only thing I can tell for now is that I only see these logs on the problematic child (and not on the others) and whenever I trash it so it recreates a pod, these logs aren't appearing on the new one.
I'll check the logs again whenever I get the alert

The dbengine database is located on the parent, the childs should be storing everything in ram, here is the childs' config →
```
bash-5.1# cat /etc/netdata/netdata.conf
[global]
  memory mode = ram
[health]
  enabled = no
[ml]
  enabled = yes
  maximum num samples to train = 3600
  minimum num samples to train = 900
  train every = 3600
  charts to skip from training = !system.* !cgroup.* *
bash-5.1#
```
The parent stores approx. 27G of data for each cluster

**Comment by MrZammler** [comment]
Well, if you are talking apo the `ACLK RES` logs, then these are connectivity messages for the cloud. Are the children claimed to the cloud, or just the parent?

**Comment by zeylos** [comment]
Sorry I didn't see the comment, the children are claimed too

**Comment by koebi001** [comment]
I also see lots of disk write operations from netdata on some of my KVM virtual machines.
The write operations go to `/var/cache/netdata/netdata-meta.db-wal` although I am using `memory mode = ram`.
All nodes are claimed to the cloud and run version `v1.34.1` installed using the `kickstart.sh` (`INSTALL_TYPE='kickstart-build'`).
There is nothing in the error log (`/var/log/netdata/error.log`) when the disk write operations start.
In the access log (`/var/log/netdata/error.log`) occasionally (like 20 per hour) I see lines including `ACLK RES` `CHARTS SENT from` which changes to nearly one per second when the disk write operations start.

**Comment by ilyam8** [comment]
> The write operations go to /var/cache/netdata/netdata-meta.db-wal

Good finding! I also see that _bug_ several times when the file keeps constantly increasing (16M->(sic!)21G)

**Comment by MrZammler** [comment]
> I also see lots of disk write operations from netdata on some of my KVM virtual machines. The write operations go to `/var/cache/netdata/netdata-meta.db-wal` although I am using `memory mode = ram`. All nodes are claimed to the cloud and run version `v1.34.1` installed using the `kickstart.sh` (`INSTALL_TYPE='kickstart-build'`). There is nothing in the error log (`/var/log/netdata/error.log`) when the disk write operations start. In the access log (`/var/log/netdata/error.log`) occasionally (like 20 per hour) I see lines including `ACLK RES` `CHARTS SENT from` which changes to nearly one per second when the disk write operations start.

Thanks @koebi001 for this!

**Comment by zeylos** [comment]
Hi,

I confirm the issue is still present with `v1.34.0-52-nightly`.
I also confirm the issue comes from the child (using `memory mode = ram`) and not the parent (if I reschedule a parent on a problematic node, the disk still writes a lot), the writes are indeed directed to `/var/cache/netdata/netdata-meta.db-wal`, although the size doesn't change (it stays at 16MB).

I'm down for a debugging session whenever you need, I currently have a bugging node on a staging cluster, it has been 3 days and it doesn't reduce

**Comment by zeylos** [comment]
Hi there,

This bug is getting pretty serious for my case, it's happening nearly everyday on a random node of each cluster, which means I need to restart at least 3 or 4 nodes everyday :(.
It's problematic for kubernetes because when this happens Kubelet detects DiskPressure and stops scheduling pods on the problematic node.

I have 2 nodes on a testing cluster that have the alert up for 2 days, if someone's up for a debugging session we can use them.
I'm available on Discord if you need anything

**Comment by MrZammler** [comment]
Hi @zeylos sorry we didn't have much news since some days...

There are a couple of PRs to look for that might fix/reduce this problem. One is https://github.com/netdata/netdata/pull/12859 which is already merged, and the other is https://github.com/netdata/netdata/pull/12852 which should go in soon.

Please let us know if after these situation has gotten better/worse/same. Thank you again for your patience!

**Comment by zeylos** [comment]
Hi there,

Updated the nodes this morning to v1.34.0-157-nightly which should contain both PRs, but still have the issue :(
![image](https://user-images.githubusercontent.com/24226163/168282133-cac74e84-9fc8-47b9-9bea-fb56504ed5a6.png)

**Comment by zeylos** [comment]
Hi there,

Good news, looks like the bug is resolved with `v1.34.0-202-nightly`, I upgraded 2 clusters (7 nodes) on Friday and no IO load since then ! :).

I'll keep a eye on it for some time and tell you if it ever happens again.

Great job guys thank you !

**Comment by MrZammler** [comment]
@zeylos thanks again for the follow up and tests!!
